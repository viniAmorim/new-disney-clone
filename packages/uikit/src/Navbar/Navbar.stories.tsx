import React from 'react';

import { Story, Meta } from '@storybook/react';

import Navbar, { NavbarProps } from '~/Navbar';

export default {
  title: 'Layout/Navbar',
  component: Navbar,
} as Meta;

const Template: Story<NavbarProps> = args => <Navbar {...args} />;

export const Default = Template.bind({});
Default.args = {
  items: [
    {
      label: 'Painel',
      link: '/painel',
      active: true,
    },
    {
      label: 'Perfil',
      link: '/perfil',
    },
  ],
};
