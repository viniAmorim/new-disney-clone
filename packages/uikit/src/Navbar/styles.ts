import styled from 'styled-components';

export const Logo = styled.img`
  width: 150px;
  height: 80px;
  padding: 0px;
`;

export const Container = styled.div`
  background-color: #13151f;
  padding: 10px 0px;
`;

export const Menu = styled.ul`
  display: flex;
  list-style: none;
  justify-content: flex-end;
`;

export const Item = styled.li`
  display: inline;
  margin: 0px 12px;
`;

interface LinkProps {
  active?: boolean;
}

export const Link = styled.a<LinkProps>`
  cursor: pointer;
  text-decoration: none;
  color: ${({ active = false, theme }) =>
    active ? theme.colors['blue-400'] : theme.colors.white};
  ${({ theme }) => theme.typography['subheader-heavy']};

  &:hover,
  &:focus,
  &:active {
    color: ${({ theme }) => theme.colors['blue-300']};
  }
`;

export const DropDownMenu = styled.ul`
  background: ${({ theme }) => theme.colors.white};
  border-radius: 5px;
  right: 0;
  list-style: none;
  padding: 15px 25px;
  position: absolute;
  top: 20px;
  width: max-content;

  margin-top: 50px;
  margin-right: 80px;

  a {
    color: ${({ theme }) => theme.colors['gray-100']};

    &:hover,
    &:focus,
    &:active {
      color: ${({ theme }) => theme.colors['blue-400']};
    }
  }
`;

export const Avatar = styled.img`
  width: 40px;
  height: 40px;
  border-radius: 50%;
  margin-left: 15px;
  margin-top: -7px;
`;

