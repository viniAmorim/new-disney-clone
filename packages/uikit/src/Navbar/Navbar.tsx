import React, { BaseHTMLAttributes } from 'react';
import { Col, Container, Row } from 'react-grid-system';

import * as Styled from './styles';

interface Item {
  label: string;
  link: string;
  active?: boolean;
}

export interface NavbarProps extends BaseHTMLAttributes<HTMLDivElement> {
  /**
   * Navbar items.
   */
  items: Item[];
}

const Navbar = ({ items, ...rest }: NavbarProps) => {
  return (
    <Styled.Container {...rest} suppressHydrationWarning>
      <Container>
        <Row align="center">
          <Col>
            <Styled.Logo src={require("./disney-plus-logo-1.png")} />
          </Col>
          <Col>
            <Styled.Menu>
              {items.map((tempItem) => (
                <Styled.Item key={tempItem.link}>
                  <Styled.Link href={tempItem.link} active={tempItem.active}>
                    {tempItem.label}
                  </Styled.Link>
                </Styled.Item>
              ))}
              <Styled.Avatar src={require("./avatar.jpg")} />
            </Styled.Menu>
          </Col>
        </Row>
      </Container>
    </Styled.Container>
  );
};

export default Navbar;
