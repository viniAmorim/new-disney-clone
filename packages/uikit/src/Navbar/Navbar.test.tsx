/* eslint-disable @typescript-eslint/no-var-requires */
import React from 'react';

import { render } from '~/test/utils';

import Navbar from './Navbar';

describe('Navbar', () => {
  it('Should render the component', () => {
    const { queryByTestId } = render(
      <Navbar
        data-testid="navbar"
        items={[
          {
            label: 'Painel',
            link: '/painel',
          },
          {
            label: 'Contatos',
            link: '/contatos',
          },
          {
            label: 'Comunicação',
            link: '/comunicacao',
          },
        ]}
      />,
    );

    const navbar = queryByTestId('navbar');

    expect(navbar).toBeInTheDocument();
    expect(navbar).toMatchSnapshot();
  });
});
