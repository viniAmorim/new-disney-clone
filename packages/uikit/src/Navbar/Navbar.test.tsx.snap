// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Should render the component 1`] = `
<div
  class="sc-eCstlR gjctib"
  data-testid="navbar"
>
  <div
    style="box-sizing: border-box; position: relative; margin-left: auto; margin-right: auto; padding-left: 15px; padding-right: 15px; max-width: 960px;"
  >
    <div
      style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: center; justify-content: flex-start; flex-direction: row;"
    >
      <div
        style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 0px; flex-grow: 1; flex-shrink: 0; max-width: 100%; margin-left: 0%;"
      >
        <svg
          fill="none"
          height="42"
          viewBox="0 0 207 42"
          width="207"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.40741 34.0132C12.8118 34.0132 15.3764 31.8086 15.8757 28.6143H14.4685C14.0146 31.1225 11.9607 32.7422 9.40741 32.7422C5.76476 32.7422 3.27958 29.4466 3.27958 25.0487C3.27958 20.5946 5.86689 17.4003 9.40741 17.4003C12.0288 17.4003 13.9919 19.1324 14.4345 21.4607H15.8416C15.331 18.2776 12.6983 16.118 9.39607 16.118C4.98176 16.118 1.90649 19.9423 1.90649 25.0712C1.90649 30.2002 4.95906 34.0132 9.40741 34.0132Z"
            fill="#FDFDFD"
          />
          <path
            d="M25.4782 34.0132C29.7904 34.0132 32.9224 30.2565 32.9224 25.0712C32.9224 19.886 29.779 16.118 25.4782 16.118C21.166 16.118 18.034 19.8748 18.034 25.0712C18.034 30.2565 21.166 34.0132 25.4782 34.0132ZM25.4782 32.7422C21.8129 32.7422 19.4071 29.3566 19.4071 25.0712C19.4071 20.8083 21.8356 17.4003 25.4782 17.4003C29.1322 17.4003 31.5493 20.7971 31.5493 25.0712C31.5493 29.3566 29.1436 32.7422 25.4782 32.7422Z"
            fill="#FDFDFD"
          />
          <path
            d="M36.0935 33.6533H37.478V22.2705C37.478 19.4811 39.5773 17.4678 42.4824 17.4678C42.8455 17.4678 43.0838 17.5127 43.2654 17.5352V16.1518C43.0498 16.1405 42.8115 16.1405 42.5618 16.1405C40.2582 16.1405 38.3177 17.3103 37.5688 19.0762H37.4439V16.3767H36.0935V33.6533Z"
            fill="#FDFDFD"
          />
          <path
            d="M45.3869 33.6533H46.7713V22.2705C46.7713 19.4811 48.8706 17.4678 51.7757 17.4678C52.1388 17.4678 52.3771 17.5127 52.5587 17.5352V16.1518C52.3431 16.1405 52.1048 16.1405 51.8551 16.1405C49.5515 16.1405 47.611 17.3103 46.8621 19.0762H46.7373V16.3767H45.3869V33.6533Z"
            fill="#FDFDFD"
          />
          <path
            d="M60.947 34.0132C64.3854 34.0132 66.4961 32.0223 67.2905 30.2789L66.0308 29.7728C65.384 31.055 63.784 32.7422 60.947 32.7422C56.9866 32.7422 54.6149 29.5703 54.5582 25.3187H67.8578V24.4639C67.8578 19.3349 64.5216 16.118 60.6179 16.118C56.2944 16.118 53.1851 19.931 53.1851 25.0937C53.1851 30.2452 56.215 34.0132 60.947 34.0132ZM54.5809 24.0702C54.8532 20.3022 57.3157 17.4003 60.6179 17.4003C63.9315 17.4003 66.4734 20.3247 66.4734 24.0702H54.5809Z"
            fill="#FDFDFD"
          />
          <path
            d="M77.188 16.3767H73.5227V12.2375H72.1496V16.3767H69.5736V17.6027H72.1496V29.8178C72.1496 32.4385 74.2036 33.9345 76.3823 33.9345C77.0859 33.9345 77.5625 33.8107 77.9937 33.6533L77.6533 32.4048C77.3129 32.5172 76.9724 32.6072 76.4731 32.6072C74.9298 32.6072 73.5227 31.6624 73.5227 29.5703V17.6027H77.188V16.3767Z"
            fill="#FDFDFD"
          />
          <path
            d="M85.3437 34.0244C88.1126 34.0244 90.0644 32.4723 90.9723 30.5376H91.0744V33.6533H92.4588V21.5844C92.4588 17.8052 89.6219 16.118 86.5239 16.118C83.8345 16.118 81.3153 17.3215 80.3166 19.8073L81.6216 20.2909C82.3025 18.5813 84.0728 17.3778 86.558 17.3778C89.3722 17.3778 91.0744 18.9862 91.0744 21.4944V22.2143C91.0744 23.5415 89.1226 23.6765 86.4672 23.9915C81.8146 24.5538 79.8173 26.016 79.8173 28.8955C79.8173 32.1798 82.3366 34.0244 85.3437 34.0244ZM85.4799 32.7535C83.0401 32.7535 81.2018 31.3025 81.2018 28.9405C81.2018 26.8034 82.7905 25.6561 86.6033 25.2062C88.3055 25.015 90.3481 24.7001 91.0744 24.1039V26.6459C91.0744 30.1552 88.7481 32.7535 85.4799 32.7535Z"
            fill="#FDFDFD"
          />
          <path
            d="M100.542 33.9458C102.97 33.9458 104.65 33.0122 105.603 31.2463H105.739V33.6534H110.959V21.9107C110.959 18.2552 107.68 16.1519 103.243 16.1519C98.5559 16.1519 95.9005 18.4914 95.458 21.6408L100.576 21.8207C100.814 20.7184 101.733 20.0436 103.197 20.0436C104.559 20.0436 105.421 20.6959 105.421 21.8545V21.9107C105.421 22.968 104.264 23.1929 101.291 23.4516C97.7616 23.7441 94.8112 25.0376 94.8112 28.8618C94.8112 32.2811 97.2055 33.9458 100.542 33.9458ZM102.255 30.3465C100.973 30.3465 100.065 29.7391 100.065 28.5919C100.065 27.4783 100.95 26.8035 102.528 26.5673C103.56 26.4211 104.831 26.1961 105.455 25.8699V27.5121C105.455 29.1993 104.026 30.3465 102.255 30.3465Z"
            fill="#0458F2"
          />
          <path
            d="M112.754 40.1321H118.303V30.8527H118.417C119.12 32.4499 120.686 33.9008 123.319 33.9008C127.177 33.9008 130.287 30.9089 130.287 25.0263C130.287 18.9301 126.996 16.1519 123.353 16.1519C120.596 16.1519 119.086 17.749 118.417 19.335H118.247V16.3768H112.754V40.1321ZM118.19 25.0151C118.19 22.1919 119.37 20.4485 121.401 20.4485C123.455 20.4485 124.59 22.2369 124.59 25.0151C124.59 27.8045 123.455 29.6154 121.401 29.6154C119.37 29.6154 118.19 27.8158 118.19 25.0151Z"
            fill="#0458F2"
          />
          <path
            d="M131.597 40.1321H137.146V30.8527H137.259C137.963 32.4499 139.529 33.9008 142.162 33.9008C146.02 33.9008 149.129 30.9089 149.129 25.0263C149.129 18.9301 145.838 16.1519 142.196 16.1519C139.438 16.1519 137.929 17.749 137.259 19.335H137.089V16.3768H131.597V40.1321ZM137.032 25.0151C137.032 22.1919 138.213 20.4485 140.244 20.4485C142.298 20.4485 143.433 22.2369 143.433 25.0151C143.433 27.8045 142.298 29.6154 140.244 29.6154C138.213 29.6154 137.032 27.8158 137.032 25.0151Z"
            fill="#0458F2"
          />
          <rect
            fill="#0458F2"
            height="24.8136"
            rx="4"
            transform="matrix(0.910102 -0.414394 0.421736 0.906724 159.278 19.501)"
            width="9.93694"
          />
          <rect
            fill="#0458F2"
            height="39.8493"
            rx="4"
            transform="matrix(0.421616 0.906779 -0.910047 0.414513 201.805 16.4568)"
            width="9.95302"
          />
          <rect
            fill="#5BE1FF"
            height="24.8136"
            rx="4"
            transform="matrix(0.910102 -0.414395 0.421736 0.906724 160.277 16.5315)"
            width="9.93693"
          />
          <rect
            fill="#5BE1FF"
            height="39.8493"
            rx="4"
            transform="matrix(0.421616 0.906779 -0.910048 0.414513 202.804 13.4873)"
            width="9.95302"
          />
        </svg>
      </div>
      <div
        style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 0px; flex-grow: 1; flex-shrink: 0; max-width: 100%; margin-left: 0%;"
      >
        <input
          class="sc-bdfBQB sc-jSgvzq kuITIP kjFrQK"
          inputmode="numeric"
          placeholder="Buscar contato"
          type="text"
          value=""
        />
      </div>
      <div
        style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 100%; flex-basis: 0px; flex-grow: 1; flex-shrink: 0; max-width: 100%; margin-left: 0%;"
      >
        <ul
          class="sc-gKseQn fZugqP"
        >
          <li
            class="sc-iBPTik fIYQLc"
          >
            <a
              class="sc-fubCzh eUoZmR"
              href="/painel"
            >
              Painel
            </a>
          </li>
          <li
            class="sc-iBPTik fIYQLc"
          >
            <a
              class="sc-fubCzh eUoZmR"
              href="/contatos"
            >
              Contatos
            </a>
          </li>
          <li
            class="sc-iBPTik fIYQLc"
          >
            <a
              class="sc-fubCzh eUoZmR"
              href="/comunicacao"
            >
              Comunicação
            </a>
          </li>
        </ul>
      </div>
    </div>
    <span
      style="display: table; clear: both;"
    />
  </div>
</div>
`;
